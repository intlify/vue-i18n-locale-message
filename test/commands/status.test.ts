import { flash, runCommand } from '../helper'

// ------
// mocks

const mockStatus = jest.fn()
jest.mock('@scope/l10n-service-provider', () => {
  return jest.fn().mockImplementation(() => {
    return { status: mockStatus }
  })
})
import L10nServiceProvider from '@scope/l10n-service-provider'

jest.mock('../../src/commands/fails/status', () => ({
  ...jest.requireActual('../../src/commands/fails/status'),
  fail: jest.fn()
}))
import statusFail from '../../src/commands/fails/status'

// --------------------
// setup/teadown hooks

let spyLog
beforeEach(() => {
  spyLog = jest.spyOn(global.console, 'log')
})

afterEach(() => {
  spyLog.mockRestore()
  jest.clearAllMocks()
})

// -----------
// test cases

test('require options', async () => {
  // mocking ...
  const mockStatusFail = statusFail as jest.Mocked<typeof statusFail>
  mockStatusFail.fail.mockImplementation(() => {})

  // run
  await runCommand('src/commands/status', `status`)
  await flash()

  // verify
  expect(mockStatusFail.fail).toHaveBeenCalled()
})

test('getTranslationStatus: done', async () => {
  // setup mocking ...
  const mockStatusFail = statusFail as jest.Mocked<typeof statusFail>
  mockStatusFail.fail.mockImplementation(() => {})
  const mockStatusValue = [{
    locale: 'en',
    percentage: 100
  }, {
    locale: 'ja',
    percentage: 100
  }]
  mockStatus.mockImplementation(() => Promise.resolve(mockStatusValue))

  // run
  await runCommand('src/commands/status', `status --provider=@scope/l10n-service-provider`)
  await flash()

  // verify
  expect(mockStatusFail.fail).not.toHaveBeenCalled()
  expect(spyLog.mock.calls[0][0]).toMatchSnapshot()
})

test('getTranslationStatus: wip', async () => {
  // setup mocking ...
  const mockStatusFail = statusFail as jest.Mocked<typeof statusFail>
  mockStatusFail.fail.mockImplementation(() => {})
  const mockStatusValue = [{
    locale: 'en',
    percentage: 72.4
  }, {
    locale: 'ja',
    percentage: 100
  }]
  mockStatus.mockImplementation(() => Promise.resolve(mockStatusValue))

  // run
  await runCommand('src/commands/status', `status --provider=@scope/l10n-service-provider`)
  await flash()

  // verify
  expect(mockStatusFail.fail.mock.calls[0][1].message).toEqual('Translation work in progress')
  expect(spyLog.mock.calls[0][0]).toMatchSnapshot()
})
