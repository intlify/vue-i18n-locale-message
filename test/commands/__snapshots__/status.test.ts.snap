// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getTranslationStatus: done 1`] = `
"┌─────────┬────────┬────────────┐
│ (index) │ locale │ percentage │
├─────────┼────────┼────────────┤
│    0    │  'en'  │    100     │
│    1    │  'ja'  │    100     │
└─────────┴────────┴────────────┘"
`;

exports[`getTranslationStatus: wip 1`] = `
"┌─────────┬────────┬────────────┐
│ (index) │ locale │ percentage │
├─────────┼────────┼────────────┤
│    0    │  'en'  │    72.4    │
│    1    │  'ja'  │    100     │
└─────────┴────────┴────────────┘"
`;
